src/sage/schemes/hyperelliptic_curves_smooth_model/hyperelliptic_constructor.py

```diff
@@ -0,0 +1,279 @@
+r"""
+Constructor for hyperelliptic curves using the smooth model
+in weighted projective space `\mathbb{P}(1 : g + 1 : 1)`.
+
+Adapted from /hyperelliptic/constructor.py
+
+AUTHORS:
+
+- David Kohel (2006): initial version
+- Sabrina Kunzweiler, Gareth Ma, Giacomo Pope (2024): adapt to smooth model
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2025 Sabrina Kunzweiler, Gareth Ma, Giacomo Pope
+#                     2019 Anna Somoza <anna.somoza.henares@gmail.com>
+#                     2006 David Kohel <kohel@maths.usyd.edu>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.categories.finite_fields import FiniteFields
+from sage.rings.abc import pAdicField
+from sage.rings.integer import Integer
+from sage.rings.polynomial.polynomial_element import Polynomial
+from sage.rings.polynomial.polynomial_ring_constructor import PolynomialRing
+from sage.rings.rational_field import RationalField
+from sage.schemes.hyperelliptic_curves_smooth_model.hyperelliptic_finite_field import (
+    HyperellipticCurveSmoothModel_finite_field,
+)
+from sage.schemes.hyperelliptic_curves_smooth_model.hyperelliptic_g2 import (
+    HyperellipticCurveSmoothModel_g2,
+    HyperellipticCurveSmoothModel_g2_finite_field,
+    HyperellipticCurveSmoothModel_g2_padic_field,
+    HyperellipticCurveSmoothModel_g2_rational_field,
+)
+from sage.schemes.hyperelliptic_curves_smooth_model.hyperelliptic_generic import (
+    HyperellipticCurveSmoothModel_generic,
+)
+from sage.schemes.hyperelliptic_curves_smooth_model.hyperelliptic_padic_field import (
+    HyperellipticCurveSmoothModel_padic_field,
+)
+from sage.schemes.hyperelliptic_curves_smooth_model.hyperelliptic_rational_field import (
+    HyperellipticCurveSmoothModel_rational_field,
+)
+
+"""
+TODO:
+
+- We currently cannot support the construction of curves over rings
+
+"""
+
+
+def HyperellipticCurveSmoothModel(
+    f, h=0, check_squarefree=True, distinguished_point=None
```

Comment:
```suggestion
    f, h=0, check_squarefree: bool = True, distinguished_point=None
```

While we don't require type annotations, they are encouraged. Especially when you're dealing with basic types where it's easy to figure out what the correct annotation is.